pkg/coredns/main.go

```diff
@@ -0,0 +1,96 @@
+package main
```

Comment:
I believe this file is taken from coredns repo, right? Would be good to add some comments explaining that:
1. Source repo
2. Git tag/branch

This is to make it easy to track which version of coredns/main.go we're using and to make sure it aligns with versions we using in plugin, which currently is 1.5.2

Reply:
Thanks @vthapar I'll add the documentation reference as a comment with some explanation.

For (2) I'm importing coredns
https://github.com/submariner-io/lighthouse/blob/2b9037545abd2c1388f1e45e9d0d1087b2da290a/pkg/coredns/main.go#L44-L45

which is the same version specified in `go.mod`
https://github.com/submariner-io/lighthouse/blob/2b9037545abd2c1388f1e45e9d0d1087b2da290a/go.mod#L7

is this what you meant?

Reply:
@vthapar the list of plugins doesn’t match the list in CoreDNS 1.5.2 exactly; is that normal? We’re missing `dns64`, `acl`, `transfer`, `azure`, `clouddns`, and `kubernetes`, and we’re adding `federation` and `sign`.

Reply:
@skitt I used `lighthouse-coredns -plugins` to compare the list on both versions

Reply:
@SteveMattar yes, this isn’t anything wrong with your changes, I’m just curious why the list is different (which would make it less useful to give the original reference for the `main.go` file).

Reply:
Looks like I was sitting on this comment. @skitt as discussed, this is okay. we will possible prune this list later as we're not really using most of these. Let us get this merged, pruning can be done later once aswin's changes to upgrade to newer version of coredns come in.